assembly/vm-mock/env.ts

```diff
@@ -30,3 +30,92 @@ export declare function mockScCall(value: StaticArray<u8>): void;
  */
 @external("massa", "assembly_script_add_address_to_ledger")
 export declare function addAddressToLedger(address: string): void;
+
+/**
+ * Emulate an admin context by giving the write access to all contracts
+ *
+ * @remarks
+ * This function is used to test write access protected smart contract functions.
+ *
+ * @example
+ * ```typescript
+ * test('test protected function', () => {
+ *   mockAdminContext();
+ *
+ *   const mockValue: StaticArray<u8> = [1,2,3];
+ *   const res = myProtectedSCFunc(NoArgs.serialize());
+ *
+ *   expect(res).toBe(mockValue);
+ * });
+ * ```
+ */
+@external("massa", "assembly_script_mock_admin_context")
+export declare function mockAdminContext(): void;
+
+/**
+ * Reset the context to the default one by removing all the write access
+ * and the deployment emulation.
+ *
+ * @example
+ * ```typescript
+ * test('test function protection', () => {
+ *   resetContext();
```

Comment:
As I understand it should be mockNonAdminContext or maybe the function should be called resetContext()